src/cache.cc

```diff
@@ -10,14 +10,20 @@
 
 namespace {
 
-std::string GetCachedBaseFileName(const std::string& cache_directory,
-                                  std::string source_file) {
-  assert(!cache_directory.empty());
-  std::replace(source_file.begin(), source_file.end(), '\\', '_');
-  std::replace(source_file.begin(), source_file.end(), '/', '_');
-  std::replace(source_file.begin(), source_file.end(), ':', '_');
-
-  return cache_directory + source_file;
+std::string GetCachedBaseFileName(Config* config,
+                                  const std::string& source_file,
+                                  bool create_dir = false) {
+  assert(!config->cacheDirectory.empty());
+  std::string cache_file;
+  size_t len = config->projectRoot.size();
+  if (source_file.substr(0, len) == config->projectRoot) {
+    // TODO Use proper filename join.
```

Comment:
Remove the TODO? I think '/' is fine for every platform. Otherwise want to add a PathJoin(a, b) function to platform.h?

Reply:
Removed // TODO